table.go

```diff
@@ -29,26 +34,34 @@ type hasSetDeletedAt interface {
 	SetDeletedAt(time.Time)
 }
 
-// Save inserts or updates given records. Auto-detects insert vs update by ID.
+// Save inserts or updates given records. Auto-detects insert vs update by ID based on zerovalue of ID from GetID() method on record.
 func (t *Table[T, PT, IDT]) Save(ctx context.Context, records ...PT) error {
-	if len(records) != 1 {
+	switch len(records) {
+	case 0:
+		return nil
+	case 1:
+		return t.saveOne(ctx, records[0])
+	default:
 		return t.saveAll(ctx, records)
 	}
+}
 
-	record := records[0]
-
+func (t *Table[T, PT, IDT]) saveOne(ctx context.Context, record PT) error {
 	if err := record.Validate(); err != nil {
-		return fmt.Errorf("save: validate record: %w", err)
+		return fmt.Errorf("validate record: %w", err)
 	}
 
 	if row, ok := any(record).(hasSetUpdatedAt); ok {
 		row.SetUpdatedAt(time.Now().UTC())
 	}
 
 	// Insert
-	var zero IDT
-	if record.GetID() == zero {
-		q := t.SQL.InsertRecord(record).Into(t.Name).Suffix("RETURNING *")
+	if record.GetID() == *new(IDT) {
```

Comment:
can we use `var zero IDT` instead? It doesn't require allocation as with `new()`